### PR TITLE
Generalizing & Refactoring

### PR DESCRIPTION
**Pull request by nikitap492**
There was too many differences in code between sort classes, so I decided to generalize it.
I've created utility class which removes some boilerplate and the general interface for the most algorithms
Also, I replaced all algorithms in a new directory, like others java projects
Fixed some bugs in the algorithms and did some perfomance improvments

### CODE REVIEW

File: Sorts/src/sort/BogoSort.java [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+
+/**
+ *
+ * @author Podshivalov Nikita (https://github.com/nikitap492)
+ *
+ * @see SortAlgorithm
+ *
+ */
+public class BogoSort implements SortAlgorithm{
```
Comment (christianbender) [inline Sorts/src/sort/BogoSort.java]: Please add space between the interface name and the curly brace.

File: Sorts/src/sort/BubbleSort.java [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+/**
+ *
+ * @author Varun Upadhyay (https://github.com/varunu28)
+ * @author Podshivalov Nikita (https://github.com/nikitap492)
+ *
+ * @see SortAlgorithm
+ */
+
+class BubbleSort implements SortAlgorithm{
```
Comment (christianbender) [inline Sorts/src/sort/BubbleSort.java]: Same thing as above.

**Comment by nikitap492** [comment]
Done

**Comment by nikitap492** [comment]
Added the new algorithms

**Pull request merged**